src/cr/cube/util.py

```diff
@@ -54,65 +42,4 @@ def fget(self):
-
-        A lazyproperty is read-only, so there is only an *fget* function (a
-        regular @property can also have an fset and fdel function). This name
-        was chosen for consistency with Python's `property` class which uses
-        this name for the corresponding parameter.
-        """
-        # ---maintain a reference to the wrapped getter method
-        self._fget = fget
-        # ---adopt fget's __name__, __doc__, and other attributes
-        functools.update_wrapper(self, fget)
-
-    def __get__(self, obj, type=None):
-        """Called on each access of 'fget' attribute on class or instance.
-
-        *self* is this instance of a lazyproperty descriptor "wrapping" the
-        property method it decorates (`fget`, nominally).
-
-        *obj* is the "host" object instance when the attribute is accessed
-        from an object instance, e.g. `obj = Obj(); obj.fget`. *obj* is None
-        when accessed on the class, e.g. `Obj.fget`.
-
-        *type* is the class hosting the decorated getter method (`fget`) on
-        both class and instance attribute access.
-        """
-        # ---when accessed on class, e.g. Obj.fget, just return this
-        # ---descriptor instance (patched above to look like fget).
-        if obj is None:
-            return self
-
-        # ---when accessed on instance, start by checking instance __dict__
-        value = obj.__dict__.get(self.__name__)
-        if value is None:
-            # ---on first access, __dict__ item will absent. Evaluate fget()
-            # ---and store that value in the (otherwise unused) host-object
-            # ---__dict__ value of same name ('fget' nominally)
-            value = self._fget(obj)
-            obj.__dict__[self.__name__] = value
-        return value
-
-    def __set__(self, obj, value):
-        """Raises unconditionally, to preserve read-only behavior.
-
-        This decorator is intended to implement immutable (and idempotent)
-        object attributes. For that reason, assignment to this property must
-        be explicitly prevented.
-
-        If this __set__ method was not present, this descriptor would become
-        a *non-data descriptor*. That would be nice because the cached value
-        would be accessed directly once set (__dict__ attrs have precedence
-        over non-data descriptors on instance attribute lookup). The problem
-        is, there would be nothing to stop assignment to the cached value,
-        which would overwrite the result of `fget()` and break both the
-        immutability and idempotence guarantees of this decorator.
-
-        The performance with this __set__() method in place was roughly 0.4
-        usec per access when measured on a 2.8GHz development machine; so
-        quite snappy and probably not a rich target for optimization efforts.
-        """
-        raise AttributeError("can't set attribute")
+    return property(functools.lru_cache()(f))
```

Comment:
I'm open to arguments on how this can be improved, but I don't think this is it. Happy to talk it over with you but in the meantime I think we should change this back to the way it was.